hook!-full-game/src/Player/States/Wall.gd

```diff
@@ -0,0 +1,39 @@
+extends State
+
+const WALL_SLIDE_SPEED: = 100.0
+const WALL_PULL_STRENGTH: = 300.0
+
+var _jump_strength: = 1000.0
+var _velocity: = Vector2.ZERO
+var _wall_direction: = -1
+
+func enter(msg: Dictionary = {}) -> void:
+	_wall_direction = msg.direction
+
+
+func physics_process(delta: float) -> void:
+	_velocity.y += WALL_SLIDE_SPEED * delta
+	_velocity.x += (WALL_PULL_STRENGTH * -_wall_direction) * delta
+	_velocity = _player.move_and_slide(_velocity, _player.FLOOR_NORMAL)
+
+	if _player.is_on_floor():
+		_state_machine.transition_to("Move/Idle")
+
+
+func unhandled_input(event: InputEvent) -> void:
+	if event.is_action_pressed("jump"):
+		var jump_velocity: = Vector2()
```

Comment:
You can write

```
		var direction: = Vector2(_wall_direction, -1.0).normalized()
		var jump_velocity: = direction * _jump_strength
```

Or 

```
		var jump_velocity: = Vector2(_wall_direction, -1.0).normalized() * _jump_strength
```

Instead of changing the variable 3 times, I think it's easier to read.